test/parallel/test-setproctitle.js

```diff
@@ -1,44 +1,73 @@
 'use strict';
-// Original test written by Jakub Lekstan <kuebzky@gmail.com>
 const common = require('../common');
 
+// Original test written by Jakub Lekstan <kuebzky@gmail.com>
+
 // FIXME add sunos support
 if (common.isSunOS) {
   return common.skip(`Unsupported platform [${process.platform}]`);
 }
 
 const assert = require('assert');
-const exec = require('child_process').exec;
+const { exec, spawn } = require('child_process');
 const path = require('path');
-
 // The title shouldn't be too long; libuv's uv_set_process_title() out of
 // security considerations no longer overwrites envp, only argv, so the
 // maximum title length is possibly quite short.
-let title = String(process.pid);
+const title = String(Date.now()).substr(-4, 4);
 
-assert.notStrictEqual(process.title, title);
-process.title = title;
-assert.strictEqual(process.title, title);
-
-// Test setting the title but do not try to run `ps` on Windows.
-if (common.isWindows) {
-  return common.skip('Windows does not have "ps" utility');
+if (process.argv[2] === 'child') {
+  const childTitle = process.argv[3];
+  assert.notStrictEqual(process.title, childTitle);
+  process.title = childTitle;
+  assert.strictEqual(process.title, childTitle);
+  // Just wait a little bit before exiting
+  setTimeout(() => {}, common.platformTimeout(300));
+  return;
 }
 
-// To pass this test on alpine, since Busybox `ps` does not
-// support `-p` switch, use `ps -o` and `grep` instead.
-const cmd = common.isLinux ?
-            `ps -o pid,args | grep '${process.pid} ${title}' | grep -v grep` :
-            `ps -p ${process.pid} -o args=`;
+// This should run only in the parent
+assert.strictEqual(process.argv.length, 2);
+let child;
+const cmd = (() => {
+  if (common.isWindows) {
+    // For windows we need to spawn a new `window` so it will get the title.
+    // For that we need a `shell`, and to use `start`
```

Comment:
Could do with a short comment here stating that changing the title on Windows changes the title of the Window, which may not be the `node` process. This add s bit of context as to why the test is different on Windows.